File: Data Structures & Algorithms/two-integer-sum/submission-4.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums: number[], target: number): number[] {
        let result: number[] = [];

        const idxmap = new Map<number, number>;
        for (let i = 0; i < nums.length; i++) {
            idxmap.set(i, nums[i]);
        }

        const mapVals: number[] = Array.from(idxmap.values());
        const mapKeys: number[] = Array.from(idxmap.keys());

        for (let i = 0; i < nums.length; i++) {
            let remaining: number = target-nums[i];
  
            if (mapVals.includes(remaining)) {
                const remainingIdx: number | undefined = mapKeys.find(key => key != i && idxmap.get(key) === remaining);

                if (remainingIdx) {
                    if (i < remainingIdx) {
                        result.push(i);
                        result.push(remainingIdx);
                        return result;
                    } else {
                        result.push(remainingIdx);
                        result.push(i);
                        return result;
                    }
                }
            }
            
        }

        return result;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)

        starts = []
        for num in nums_set:
            if num-1 not in nums_set:
                starts.append(num)

        result = 0
        for num in nums:
            if num in starts:
                i = num
                while i in nums_set:
                    i += 1

                if i-num > result:
                    result = i-num
        
        return result
                



        
        